### PR TITLE
SAK-46657 provide better error message when points dont match categor…

### PR DESCRIPTION
**Pull request by ottenhoff**
…y rules

### CODE REVIEW

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AuthorMessages.properties [hunk 1]
```diff
@@ -651,6 +651,7 @@ append_copy_title=- Copy #
 # gradebook can not accept an assessment if its totalpoint = 0, 
 gradebook_exception_min_points=Warning: A Gradebook item cannot be created for an assessment worth 0 points. Either edit the assessment to add a point value to at least one question or uncheck the "Send assessment score to Gradebook" box in Settings.
 gradebook_exception_title_invalid=Warning: Assessments configured to send scores to Gradebook cannot have titles beginning with reserved characters * ] [ or # and must be revised. Please edit the assessment title before publishing.
+gradebook_exception_category_invalid=Warning: all items in a Gradebook Category using drop lowest or drop highest must have matching point values. Either edit the assessmet to alter the point values or choose a different category.
```
Comment (adrianfish) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AuthorMessages.properties:654]:
```suggestion
gradebook_exception_category_invalid=Warning: all items in a Gradebook Category using drop lowest or drop highest must have matching point values. Either edit the assessment to alter the point values or choose a different category.
```

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/PublishAssessmentListener.java [hunk 2]
```diff
@@ -324,7 +325,13 @@ private void publish(AssessmentFacade assessment,
                                                  "gradebook_exception_title_invalid");
         FacesContext.getCurrentInstance().addMessage(null, new FacesMessage(err));
         throw new AbortProcessingException(gbe);
-    } catch (Exception e) {
+    } catch (InvalidCategoryException gbe) {
+		log.warn(gbe.getMessage());
+		String err=(String)ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AuthorMessages",
```
Comment (adrianfish) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/PublishAssessmentListener.java:330]:
```suggestion
		String err = (String)ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AuthorMessages",
```

**Review by adrianfish — CHANGES_REQUESTED** [review]

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AuthorMessages.properties (hunk 1 above)
Comment (ottenhoff) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AuthorMessages.properties:654]: nice

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/PublishAssessmentListener.java [hunk 3]
```diff
@@ -324,6 +325,12 @@ private void publish(AssessmentFacade assessment,
                                                  "gradebook_exception_title_invalid");
         FacesContext.getCurrentInstance().addMessage(null, new FacesMessage(err));
         throw new AbortProcessingException(gbe);
+    } catch (InvalidCategoryException gbe) {
+		log.warn(gbe.getMessage());
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/PublishAssessmentListener.java:329]: More informative log message?
```suggestion
		log.warn("Incorrect gradebook category settings: {}", gbe.toString());
```

**Review by adrianfish — APPROVED** [review]

**Pull request merged**